src/network/connection.cpp

```diff
@@ -126,16 +124,16 @@ void makeSplitPacket(const SharedBuffer<u8> &data, u32 chunksize_max, u16 seqnum
 		memcpy(&chunk[chunk_header_size], &data[start], payload_size);
 
 		chunks->push_back(chunk);
-		chunk_count++;
 
 		start = end + 1;
+		sanity_check(chunk_num < 0xFFFF); // overflow
```

Comment:
Couldn't this error be triggered by too large media files? I don't think it's reasonable to abort the server for such reason.
Proposal: log a warning and clear `chunks`.

Reply:
Yes. The theoretical biggest split packet (at 512B) is 32MB.

This is not new however, the sanity_check in ` UDPPeer::processReliableSendCommand` would trigger earlier than that.
Just dropping the packet would also cause media transfer on the client to stall with no apparent reason.

Reply:
Makes sense. Thanks for the explanation.